### PR TITLE
Add topic resource support, third trial

### PR DESCRIPTION
**Pull request by eliobischof**
## Description

Cherry picked the commits from #832 that has a messed up commit history. 

## PR Checklist

- [x] Resource attributes match 1:1 the names and structure of the API resource in [the GitLab API documentation](https://docs.gitlab.com/ee/api/).
- [x] [Examples](/examples) are updated with:
    - A \*.tf file for the resource/s with at least one usage example
    - A \*.sh file for the resource/s with an import example (if applicable)
- [x] New resources have at minimum a basic test with three steps:
    - Create the resource
    - Update the attributes
    - Import the resource
- [x] No new `//lintignore` comments that came from copied code. Linter rules are meant to be enforced on new code.

### CODE REVIEW

**Comment by timofurrer** [comment]
Thanks @eliobischof for creating yet another PR for this! 🎉 

I'm still super unhappy about the tradeoffs we had to make here (absolutely not your fault @eliobischof :D )- but for the lack of a better solution I think we can merge this.

I think I'll follow up with some changes in go-gitlab to support removing the avatar (since https://docs.gitlab.com/ee/api/topics.html#remove-a-topic-avatar) is available and also include the `total_projects_count` and `avatar_url` as `Computed` attributes. But it shouldn't block that PR from finally going through :)

**Review by timofurrer — APPROVED** [review]

File: examples/resources/gitlab_topic/import.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# You can import a topic to terraform state using `terraform import <resource> <id>`.
+# The `id` must be an integer for the id of the topic you want to import,
+# for example:
+terraform terraform import gitlab_topic.functional-programming 1
```
Comment (armsnyder) [inline examples/resources/gitlab_topic/import.sh:4]: Let's use underscore in name. Also double terraform typo.
```suggestion
terraform import gitlab_topic.functional_programming 1
```

File: examples/resources/gitlab_topic/resource.tf [hunk 2]
```diff
@@ -0,0 +1,4 @@
+resource "gitlab_topic" "functional-programming" {
```
Comment (armsnyder) [inline examples/resources/gitlab_topic/resource.tf:1]: Style - underscores in names.

```suggestion
resource "gitlab_topic" "functional_programming" {
```

**Review by armsnyder — DISMISSED** [review]
Looks good! Just some typos and style things.

**Comment by timofurrer** [comment]
I took the liberty and applied @armsnyder suggestions. Once the checks pass I'll merge!

Thanks @eliobischof for your effort ❤️  much appreciated 🎉

**Pull request merged**